libsolidity/codegen/ir/IRLValue.h

```diff
@@ -83,16 +86,22 @@ class IRStorageItem: public IRLValue
 	IRStorageItem(
 		IRGenerationContext& _context,
 		std::string _slot,
-		unsigned _offset,
+		boost::variant<std::string, unsigned> const& _offset,
```

Comment:
Probably better to pass in by value and move? And maybe making this one private and adding two public constructors - one for string and one for unsigned int would be nicer, maybe not, I'm not sure.

Reply:
What advantage would having two c.tors have? I mean you can pass both `unsigned` and `string` to one right now..

Reply:
In any case you should still pass by value and move instead of having a const reference here... then it's probably fine to keep one constructor, yes.

Reply:
hmm so my current `move` is actually bad if passed by `const&`?

Reply:
Not "bad", but it does nothing :-). You can't move from a ``const&``, it just stays a ``const&`` and gets copied anyways.

Reply:
pushed update